src/main/java/httpServer/http/HTTPResponseWriter.java

```diff
@@ -0,0 +1,45 @@
+package httpServer.http;
+
+import java.util.List;
+
+import static httpServer.http.Constants.CRLF;
+
+public class HTTPResponseWriter {
+
+    public String buildResponse(StatusCodes statusCode, List methods, HTTPRequest request){
+        StringBuilder responseStringBuilder = new StringBuilder();
+        responseStringBuilder.append(makeStatusLine(statusCode, request));
+        if (!statusCode.equals(StatusCodes.PAGE_NOT_FOUND)) {
+            responseStringBuilder.append(makeAllowLine(methods));
+        }
+        responseStringBuilder.append(makeContentLengthLine());
+        return responseStringBuilder.toString();
+    }
+
+    private String makeStatusLine(StatusCodes statusCode, HTTPRequest request) {
+        StringBuilder statusLine = new StringBuilder();
+        statusLine.append(request.httpVersionNumber + " ");
+        statusLine.append(statusCode.httpResponse);
+        statusLine.append(CRLF);
+        return statusLine.toString();
```

Comment:
Do you really need a `StringBuilder` here? You do a pretty nice job of just using string concatenation in your tests. Personally I find that easier to read.